graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,144 @@
+from scipy.stats import multiscale_graphcorr
+from joblib import Parallel, delayed
 
 
 class LatentDistributionTest(BaseInference):
     """
-    Two-sample hypothesis test for the problem of determining whether two random 
+    Two-sample hypothesis test for the problem of determining whether two random
     dot product graphs have the same distributions of latent positions.
-    
+
     This test can operate on two graphs where there is no known matching between
-    the vertices of the two graphs, or even when the number of vertices is different. 
+    the vertices of the two graphs, or even when the number of vertices is different.
     Currently, testing is only supported for undirected graphs.
 
     Read more in the :ref:`tutorials <inference_tutorials>`
 
     Parameters
     ----------
+    test : str, one of ["CCA", "Dcorr", "HHG", "RV", "Hsic", "MGC"]
+        Independence test to use.
+
+    distance : function, (default=`function <euclidean>`)
+        Distance function to use.
+
     n_components : int or None, optional (default=None)
         Number of embedding dimensions. If None, the optimal embedding
         dimensions are found by the Zhu and Godsi algorithm.
 
     n_bootstraps : int (default=200)
         Number of bootstrap iterations.
 
-    bandwidth : float, optional (default=0.5)
-        Bandwidth to use for gaussian kernel. If None,
-        the median heuristic will be used.
+    num_workers : int, optional (default=1)
+        Number of workers to use. If more than 1, parallelizes the code.
 
     Attributes
     ----------
     sample_T_statistic_ : float
-        The observed difference between the embedded latent positions of the two 
+        The observed difference between the embedded latent positions of the two
         input graphs.
 
-    p_ : float
+    p_value_ : float
         The overall p value from the test.
-    
-    null_distribution_ : ndarray, shape (n_bootstraps, )
+
+    null_distribution_ : None or ndarray, shape (n_bootstraps, )
         The distribution of T statistics generated under the null.
 
     References
     ----------
-    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017). 
+    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017).
         "A nonparametric two-sample hypothesis testing problem for random graphs."
         Bernoulli, 23(3), 1599-1630.
+
+    .. [2] Panda, S., Palaniappan, S., Xiong, J., Bridgeford, E., Mehta, R., Shen, C., & Vogelstein, J. (2019).
+        "hyppo: A Comprehensive Multivariate Hypothesis Testing Python Package."
+        arXiv:1907.02088.
     """
 
-    def __init__(self, n_components=None, n_bootstraps=200, bandwidth=None):
+    def __init__(
+        self,
+        test,
+        distance=euclidean,
+        n_components=None,
+        n_bootstraps=200,
+        num_workers=1,
+    ):
+        supported_tests = ["CCA", "Dcorr", "HHG", "RV", "Hsic", "MGC"]
```

Comment:
since your checks will be case sensitive, my vote would be all lower case (as opposed to upper/lower in the same string)

Reply:
agreed